beam/__init__.py

```diff
@@ -54,54 +58,259 @@ class SupportedUsageScope(Enum):
+        title="Available subcommands",
+        required=True,
+    )
+
+    beam = subparsers.add_parser(
+        "beam",
+        help="Extracts data from the provided <input_path> of type <input_type>.",
+        description=f"""
+            Takes an input_path and an input_type, installs and runs an appropriate
+            extractor, if available, from the chosen registry (default: {REGISTRY_BASE_URL}).
+        """,
+    )
+
+    beam.add_argument(
+        "input_type",
         help="FileType.ID of the input file",
-        default=None,
     )
 
-    argparser.add_argument(
-        "infile",
+    beam.add_argument(
+        "input_path",
         help="Path of the input file",
-        default=None,
     )
 
-    argparser.add_argument(
-        "--outfile",
+    beam.add_argument(
+        "--output_path",
         "-o",
         help="Optional path of the output file",
         default=None,
     )
+    beam.set_defaults(func=extract)
+
+    probe = subparsers.add_parser(
+        "probe",
+        help="Searches the registry for Extractors available for <input_type>.",
+        description=f"""
+            Searches the chosen registry (default: {REGISTRY_BASE_URL}) for Extractors
+            matching the provided input_type and returns the full definition of that
+            Extractor.
+        """,
+    )
+
+    probe.add_argument(
+        "input_type",
+        help="FileType.ID for which available Extractors are to be looked up.",
+    )
+    probe.set_defaults(func=fetch_registered_extractors)
+
+    yard = subparsers.add_parser(
+        "yard",
+        help="Searches the registry for an Extractors matching <extractor_id>.",
+        description=f"""
+            Searches the chosen registry (default: {REGISTRY_BASE_URL}) for an Extractor
+            that matches the provided extractor_id and returns its full definition.
+        """,
+    )
 
-    args = argparser.parse_args()
+    yard.add_argument(
+        "extractor_id",
+        help="Extractor.ID of the requested extractor.",
+    )
+    yard.set_defaults(func=search_extractor)
+
+    install = subparsers.add_parser(
+        "install",
+        help="Installs the Extractor matching <extractor_id>.",
+        description=f"""
+            Searches the chosen registry (default: {REGISTRY_BASE_URL}) for an Extractor
+            that matches the provided extractor_id and installs it into the current
+            Python env.
+        """,
+    )
 
-    extract(
-        input_path=args.infile,
-        input_type=args.filetype,
-        output_path=args.outfile,
-        preferred_mode=SupportedExecutionMethod.CLI,
+    install.add_argument(
+        "extractor_id",
+        help="Extractor.ID of the requested extractor.",
     )
+    install.add_argument(
+        "--use_venv",
+        action="store_true",
+        help="Install into a separate venv.",
+        default=False,
+    )
+    install.set_defaults(func=install_extractor)
```

Comment:
This is the opposite of what happens right now, right? I think the default should be to make a fresh venv, and then there can be some override to specify a specific venv, with default value of the currently loaded venv/Python, but even this could be added later.

```suggestion
    install.add_argument(
        "--use_venv",
        action="store_true",
        help="Install into a separate venv.",
        default=True,
    )
    install.set_defaults(func=install_extractor)
```

Reply:
Fair enough. Changed in the current version.